packages/sync-actions/test/customer-group-sync.spec.js

```diff
@@ -1,38 +1,136 @@
 
-describe('Exports', () => {
+describe('Discount Groups Exports', () => {
   it('action group list', () => {
-    expect(actionGroups).toEqual(['base'])
+    expect(actionGroups).toEqual(['base', 'custom'])
   })
 
-  it('correctly defined based actions list', () => {
-    expect(baseActionsList).toEqual([
-      { action: 'changeName', key: 'name' },
-      { action: 'setKey', key: 'key' },
-    ])
+  describe('action list', () => {
+    it('should contain `changeName` action', () => {
+      expect(baseActionsList).toEqual(
+        expect.arrayContaining([{ action: 'changeName', key: 'name' }])
+      )
+    })
+
+    it('should contain `setKey` action', () => {
+      expect(baseActionsList).toEqual(
+        expect.arrayContaining([
+          {
+            action: 'setKey',
+            key: 'key',
+          },
+        ])
+      )
+    })
   })
 })
 
-describe('Actions', () => {
+describe('Discount Groups Actions', () => {
   let customerGroupSync
   beforeEach(() => {
     customerGroupSync = customerGroupSyncFn()
   })
 
-  it('should build `changeName` action', () => {
-    const before = { name: 'the-greatest-name-before' }
-    const now = { name: 'the-greatest-name-now' }
-    const expected = [{ action: 'changeName', ...now }]
+  it('should build the `changeName` action', () => {
+    const before = {
+      name: { en: 'en-name-before', de: 'de-name-before' },
+    }
+
+    const now = {
+      name: { en: 'en-name-now', de: 'de-name-now' },
+    }
+
+    const expected = [
+      {
+        action: 'changeName',
+        name: { en: 'en-name-now', de: 'de-name-now' },
+      },
+    ]
     const actual = customerGroupSync.buildActions(now, before)
     expect(actual).toEqual(expected)
   })
 
-  it('should build `setKey` action', () => {
-    const before = { key: 'unique-key-before' }
-    const now = { key: 'unique-key-now' }
-    const expected = [{ action: 'setKey', ...now }]
+  it('should build the `setKey` action', () => {
+    const before = {
+      key: 'foo-key',
+    }
+
+    const now = {
+      key: 'bar-key',
+    }
+
+    const expected = [
+      {
+        action: 'setKey',
+        key: 'bar-key',
+      },
+    ]
+    const actual = customerGroupSync.buildActions(now, before)
+    expect(actual).toEqual(expected)
+  })
+
+  describe('custom fields', () => {
+    it('should build `setCustomType` action', () => {
+      const before = {
+        custom: {
+          type: {
+            typeId: 'type',
+            id: 'customType1',
+          },
+          fields: {
+            customField1: true,
+          },
+        },
+      }
+      const now = {
+        custom: {
+          type: {
+            typeId: 'type',
+            id: 'customType2',
+          },
+          fields: {
+            customField1: true,
+          },
+        },
+      }
+      const actual = customerGroupSync.buildActions(now, before)
+      const expected = [{ action: 'setCustomType', ...now.custom }]
```

Comment:
whoa... This is possible on the API? to change the custom-type __entirely__?
I wonder what would happen to the values.

Reply:
I would assume it only serves as a blueprint for new entities from there on out not for existing ones.